java/ql/src/experimental/Security/CWE/CWE-522/UnsecureBasicAuth.ql

```diff
@@ -0,0 +1,228 @@
+/**
+ * @name Unsecure basic authentication
+ * @description Basic authentication only obfuscates username/password in Base64 encoding, which can be easily recognized and reversed. Transmission of sensitive information not over HTTPS is vulnerable to packet sniffing.
+ * @kind problem
+ * @id java/unsecure-basic-auth
+ * @tags security
+ *       external/cwe-522
+ *       external/cwe-319
+ */
+
+import java
+import semmle.code.java.frameworks.Networking
+import semmle.code.java.dataflow.TaintTracking
+import DataFlow::PathGraph
+
+/**
+ * The Java class `org.apache.http.client.methods.HttpRequestBase`. Popular subclasses include `HttpGet`, `HttpPost`, and `HttpPut`.
+ * And the Java class `org.apache.http.message.BasicHttpRequest`.
+ */
+class ApacheHttpRequest extends RefType {
+  ApacheHttpRequest() {
+    hasQualifiedName("org.apache.http.client.methods", "HttpRequestBase") or
```

Comment:
Consider adding `getASourceSupertype*()` to also include sub classes of `HttpRequestBase`. 
```
this.getASourceSupertype*().hasQualifiedName("org.apache.http.client.methods", "HttpRequestBase") or
this.getASourceSupertype*().hasQualifiedName("org.apache.http.message", "BasicHttpRequest") or

Reply:
Thanks @aibaars. I've made the change.